### PR TITLE
Preview tools infrastructure setup

### PR DESCRIPTION
**Pull request by rart**

### CODE REVIEW

File: ui/app/src/modules/Preview/Tools/SimulatorPanel.tsx [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import { defineMessages, useIntl } from 'react-intl';
-import React, { useEffect, useMemo, useState } from 'react';
-import { fromString } from '../../../utils/xml';
+import React, { useEffect, useMemo, useReducer } from 'react';
```
Comment (sumerjabri) [inline ui/app/src/modules/Preview/Tools/SimulatorPanel.tsx:19]: Copyright please.

File: ui/app/src/modules/Preview/Tools/ToolPanel.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { selectTool, usePreviewContext } from '../previewContext';
-import { useIntl } from 'react-intl';
-import React from 'react';
+import { MessageDescriptor, useIntl } from 'react-intl';
```
Comment (sumerjabri) [inline ui/app/src/modules/Preview/Tools/ToolPanel.tsx:19]: Copyright

File: ui/app/src/utils/i18n.ts [hunk 3]
```diff
@@ -1,3 +1,3 @@
-export function getTranslation(key: string, table: any) {
-  return table[key] || { id: 'translationNotAvailable', defaultMessage: key };
+export function getTranslation(key: string, table: any, formatMessage = (descriptor) => descriptor) {
```
Comment (sumerjabri) [inline ui/app/src/utils/i18n.ts:18]: Copyright

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please add copyright to all files. I marked a few, but there are more.

**Comment by rart** [comment]
@sumerjabri done.

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**